Library/Homebrew/formula_installer.rb

```diff
@@ -1145,19 +1145,19 @@ def pour
     keg = Keg.new(formula.prefix)
     Tab.clear_cache
 
+    tabfile = keg/Tab::FILENAME
+    bottle_json_path = formula.local_bottle_path&.sub(/\.tar\.gz$/, ".json")
     tab = if (tab_attributes = formula.bottle_tab_attributes.presence)
-      Tab.from_file_content(tab_attributes.to_json, keg/Tab::FILENAME)
+      Tab.from_file_content(tab_attributes.to_json, tabfile)
+    elsif !tabfile.exist? && bottle_json_path&.exist?
+      _, tag, = Utils::Bottles.extname_tag_rebuild(formula.local_bottle_path)
```

Comment:
May make sense to extract all of this (perhaps including the logic above) into `utils/bottles`?